ports/espressif/common-hal/canio/CAN.c

```diff
@@ -16,7 +16,15 @@
 
 #include "hal/twai_types.h"
 
-static bool reserved_can;
+static byte reserved_can_instance; /* Bit assigned instances    0x01 (CAN0)  or 0x02 (CAN1)    0x00-NoneAssigned     0x03-Both Assigned*/
+
+#define TWAI0              REG_TWAI_BASE(0) /**< \brief (CAN0) APB Base Address */
+#define TWAI1              REG_TWAI_BASE(1) /**< \brief (CAN1) APB Base Address */
+#define TWAI_INST_NUM      2                          /**< \brief (CAN) Number of instances */
+
+static twai_handle_t twai_bus_0;
+static twai_handle_t twai_bus_1;
+static twai_handle_t *const twai_insts[TWAI_INST_NUM] = { &twai_bus_0, &twai_bus_1 };
```

Comment:
Guard this based on some defines the IDF provides for the number of TWAI. Otherwise, chips with only 1 instance won't work.